ui/src/main/java/edu/wpi/grip/ui/MainWindowView.java

```diff
@@ -190,6 +189,18 @@ public boolean saveProjectAs() throws IOException {
     }
 
     @FXML
+    public void showProjectSettingsEditor() {
+        final ProjectSettings projectSettings = pipeline.getPipeline().getProjectSettings().clone();
```

Comment:
Explain why this is cloned? Also, why is this done instead of using a copy constructor?